api/src/test/java/org/openmrs/module/fgh/mpi/MpiHttpClientTest.java

```diff
@@ -1,11 +1,184 @@
 package org.openmrs.module.fgh.mpi;
 
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.*;
+import static org.openmrs.module.fgh.mpi.MpiConstants.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ Context.class, MpiUtils.class, MpiContext.class, URL.class })
 public class MpiHttpClientTest {
 	
+	@Mock
+	private MpiHttpClient mockMpiHttpClient;
```

Comment:
This is bad practice, you should never mock the very class you are testing because then you are testing nothing, you can at most only stub out some methods if necessary.